app-experimental.tex

```diff
@@ -2693,6 +2693,125 @@ \section{Distributed Capabilities For Peripherals And Accelerators} % <<<
 
 % >>>
 % >>>
+\section{Thread Identification} % <<<
+
+\subsection{Motivation}
+
+Compartmentalisation models often rely on trusted code performing the switch from one compartment to another.
+This trusted code needs a mechanism to retrieve a data capability, e.g., for a stack.
+When using sentries (sealed entry capabilities) to call into trusted code, the same piece of trusted code can be called from multiple threads and each thread must have a distinct trusted data capability.
```

Comment:
I think more properly, it’s that each thread must have a distinct stack -- even if it’s reached via the same trusted data capability. But they will need to select the right stack.

Reply:
Hmm. Maybe in fact my comment is more broad in scope: “trusted data capability” is not a term we’ve defined previously, and may require some discussion. Note that in, for example, an otype-type model, you may already have a trustworthy data capability -- you just need a thread ID to index it so you can load the right per-thread stack capability. Whereas in other models, such as executive-restrictive, you need a dedicated stack pointer capability and don’t need to derive that from another capability.

Reply:
The term "trusted data capability" should be defined. I wanted to keep the terminology broad. Some models might not only need a trusted stack. As I understand the current form of the library-compartmentalisation model has a table of stacks for the compartments called into. This table is stored in a separate capability. One could design that to also be accessible in a thread ID indexed way.